recipes/views.py

```diff
@@ -84,4 +88,93 @@ class CollectionDeleteView(DeleteView):
     success_url = reverse_lazy('recipe:collection_list')
 
     def get_queryset(self):
-        return Collection.objects.filter(author=self.request.user)
\ No newline at end of file
+        return Collection.objects.filter(author=self.request.user)
+    
+    
+@login_required
+def create_recipe(request):
```

Comment:
View is too large. Violates clean code. Use a class-based view and split code into small meaningful methods.